modules/dicom_archive/php/dicom_archive.class.inc

```diff
@@ -75,7 +75,16 @@ class Dicom_Archive extends \DataFrameworkMenu
         $site_names   = array_values(\Utility::getSiteList());
         $site_options = array_combine($site_names, $site_names);
 
-        return [ 'sites' => $site_options ];
+        // Populate options for Sex in DICOM Archive.
+        // Note that these values differ from the ones used in Candidate
+        // as these values are extracted from imaging headers.
+        $sex_values  = ['F', 'M', 'O'];
+        $sex_options = array_combine($sex_values, $sex_values);
```

Comment:
There's no reason to change the PHP just to return a constant array which is created in a confusing way.

Reply:
It's a common idiom in LORIS at this point when creating values for a front-end dropdown.

> git grep 'array_combine'
modules/behavioural_qc/php/behavioural_qc.class.inc:        $visit_array   = array_combine($visit_array, $visit_array);
modules/dicom_archive/php/dicom_archive.class.inc:        $site_options = array_combine($site_names, $site_names);
modules/document_repository/php/document_repository.class.inc:        $instrumentsList = array_combine($instruments, $instruments);
modules/document_repository/php/files.class.inc:        $instrumentsList = array_combine($instruments, $instruments);
modules/electrophysiology_browser/php/electrophysiology_browser.class.inc:            $list_of_sites = array_combine($siteNames, $siteNames);
modules/imaging_uploader/ajax/read_log.php:    $files = array_combine($files, array_map("filemtime", $files));
modules/publication/ajax/FileUpload.php:        $currentCollabs = array_combine(
modules/publication/ajax/FileUpload.php:        $submittedCollaborators = array_combine(
modules/publication/ajax/getData.php:    $allVOIs['Behavioral'] = array_combine($bvlVOIs, $bvlVOIs);
modules/publication/ajax/getData.php:    $allVOIs['Imaging'] = array_combine($imgVOIs, $imgVOIs);
modules/publication/ajax/getData.php:    $kws = array_combine($kws, $kws);
modules/publication/ajax/getData.php:    $collabs = array_combine($collabs, $collabs);

It's used to the point where I created a generic function to do this in another PR but it didn't get merged (https://github.com/aces/Loris/pull/5328/files#diff-ebd65470c97636299b57db3ab5e1f5a8R984).

Reply:
The value being returned here is constant and only being done to pass the hardcoded value `{ M: "M", F: "F", O: "O" }` to javascript. Just code the value in javascript.

Reply:
@johnsaigle the following might be helpful
```
const dropdownArray = ['a','b','c'];
const result = dropdownArray.reduce((a,b)=> (a[b]=b,a),{});
console.log(result); // {a: "a", b: "b", c: "c"}
```

Reply:
@maltheism it is a hardcoded constant that will never change. There is no reason to use an array reduce. The value `{ M: "M", F: "F", O: "O" }` should just be hardcoded in the React.